docs/gensidebar.py

```diff
@@ -5,86 +5,91 @@
 
 import os
 
+
 def write_if_changed(fname, contents):
-    
+
     try:
-        with open(fname, 'r') as fp:
+        with open(fname, "r") as fp:
             old_contents = fp.read()
     except:
-        old_contents = ''
-        
+        old_contents = ""
+
     if old_contents != contents:
-        with open(fname, 'w') as fp:
+        with open(fname, "w") as fp:
             fp.write(contents)
 
+
 def generate_sidebar(conf, conf_api):
-    
+
     # determine 'latest' or 'stable'
     # if not conf.do_gen:
-    do_gen = os.environ.get('SIDEBAR', None) == '1' or conf['on_rtd']
-    version = conf['rtd_version']
-    
-    lines = [
-        '', '.. DO NOT MODIFY! THIS PAGE IS AUTOGENERATED!', ''
-    ]
-    
+    do_gen = os.environ.get("SIDEBAR", None) == "1" or conf["on_rtd"]
+    version = conf["rtd_version"]
+
+    lines = ["", ".. DO NOT MODIFY! THIS PAGE IS AUTOGENERATED!", ""]
+
     def toctree(name):
-        lines.extend(['.. toctree::',
-                      '    :caption: %s' % name,
-                      '    :maxdepth: 2',
-                      ''])
-    
+        lines.extend(
+            [".. toctree::", "    :caption: %s" % name, "    :maxdepth: 2", ""]
```

Comment:
does black require these to be on one line? I think it looks better when each item is on its own line

Reply:
Black isn't really configurable, except by turning formatting off. I agree this part is ugly, but it's not really user-facing so I'm fine with that.

Reply:
The alternative is to call `lines.append` a bunch of times, I guess. Follow your heart, I suppose.